packages/components/src/public/QueryModel/QueryModel.ts

```diff
@@ -1093,4 +1138,50 @@ export class QueryModel {
     }
 }
 
-type QueryModelURLState = Pick<QueryModel, 'filterArray' | 'offset' | 'schemaQuery' | 'selectedReportId' | 'sorts'>;
+type QueryModelURLState = Pick<
+    QueryModel,
+    'filterArray' | 'maxRows' | 'offset' | 'schemaQuery' | 'selectedReportId' | 'sorts'
+>;
+type QueryModelSettings = Partial<Pick<QueryModel, 'filterArray' | 'maxRows' | 'sorts' | 'viewName'>>;
+const LOCAL_STORAGE_PREFIX = 'QUERY_MODEL_SETTINGS.';
+
+function localStorageKey(modelId: string): string {
+    return LOCAL_STORAGE_PREFIX + modelId;
+}
+
+export function getSettingsFromLocalStorage(id: string): QueryModelSettings {
+    const savedSettings = JSON.parse(localStorage.getItem(localStorageKey(id)));
+
+    if (savedSettings === null) return undefined;
+
+    const { maxRows, viewName } = savedSettings;
+    const filterArray = savedSettings.filterArray?.map(f =>
+        Filter.create(f.columnName, f.value, Filter.getFilterTypeForURLSuffix(f.type))
+    );
+    const sorts = savedSettings.sorts?.map(s => querySortFromString(s));
+
+    return {
+        filterArray: filterArray ?? [],
+        maxRows: maxRows ?? DEFAULT_MAX_ROWS,
+        sorts: sorts ?? [],
+        viewName,
+    };
+}
+
+export function saveSettingsToLocalStorage(model: QueryModel): void {
+    const settings = {
+        filterArray: model.filterArray.map(f => ({
+            columnName: f.getColumnName(),
+            value: f.getValue(),
+            type: f.getFilterType().getURLSuffix(),
+        })),
+        maxRows: model.maxRows,
+        sorts: model.sorts.map(sort => sort.toRequestString()),
```

Comment:
is it possible that model.filterArray or model.sorts are null/undefined and would need a null check here? I assume not.

Reply:
We should not need a null check here, the QueryModel constructor sets default values to empty arrays for these attributes. If they are null or undefined we'd have a ton of other issues elsewhere.